app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -330,8 +311,9 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
             }
         }
 
-        viewLifecycleOwner.lifecycleScope.allowUndo(
-            requireView(), message,
+        MainScope().allowUndo(
+            requireActivity().getRootView()!!,
```

Comment:
I think we should avoid `!!` as it could cause crashes, maybe let? :)

Reply:
Hm, that's a good question! 

I don't think getRootView could actually fail after requireActivity was called successfully. The code goes like this:
`asActivity()?.window?.decorView?.findViewById<View>(android.R.id.content) as? ViewGroup`
So, `window` is there after the activity has been created and never set to null again. `decorView` is not null, and there is always a `android.R.id.content` in the decor view.

We actually have `getRootView()!!` [already](https://searchfox.org/mozilla-mobile/source/fenix/app/src/main/java/org/mozilla/fenix/library/bookmarks/edit/EditBookmarkFragment.kt#220) used with a FenixSnackbar, and I couldn't find any crashes related to that!

Going with let would not execute the removal in case there is no proper view provided by `rootView`. Hm, may be we could provide an alternative, like a fragment view? Like `requireActivity().getRootView() ?: requiteView()`
But in that case having no access to the fragment view will also have the same affect.

I think, relying on `activity.window` on the main thread is a as safe bet as it gets. But I don't have a strong opinion on that, want to here your feedback.

Reply:
Ok, that make sense. As we don't have a better alternative, let's keep it!